api-server/README.md

```diff
@@ -0,0 +1,112 @@
+
+#### The blockchain scanner daemon
+
+The blockchain scanner daemon is a tool that runs in the backend, scans the blockchain, and updates the database information.
+
+#### How the API web server and the blockchain scanner daemon work together
+
+The architecture of the API server is made to be distributed as much as desired. You can run the database on as many servers as you wish in master-slave mode. This is achieved by separating the "API web server" from the "blockchain scanner daemon". You can have a single "blockchain scanner daemon", communicating with the `node-daemon` of Mintlayer, collecting information about new blocks and writing it to the master database, while having as many instances of the API web server reading from the slave databases. This ensures virtually an infinitely scalable infrastructure.
+
+### Using other database infrastructures
+
+Currently, the API server uses PostgreSQL for storage, but the design is extremely flexible and any desired database can be added if need by implementing some interface (trait) in the rust code.
+
+In addition to the PostgreSQL, an implementation of a full in-memory storage exists, which we use for testing and as a reference implementation. Hence, when adding a new database implementation, the in-memory implementation can be used as a reference implementation. Our tests ensure that both PostgreSQL and in-memory implementation, through the beautiful abstractions of rust, both arrive to the same result. Any additional implementation can be added to the same test suite.
+
+## How to run
+
+In the following we present the minimal requirements to run the API server in action. In this example, we will be using the testnet. Replace every `testnet` with `mainnet` for the mainnet.
+
+### Make sure you have a node running
+
+If you don't have a node running, the API server won't have a source, from which it can read block data. To run the node, please consult the [the main readme file of the repo](/README.md).
+
+### How to run the database
+
+To run the database, you can use use docker or podman. For simplicity, so that root isn't needed, podman can be used:
+
+```
+podman run --detach --rm --name MintlayerAPIServerDB -e POSTGRES_HOST_AUTH_METHOD=trust -e POSTGRES_DB=mintlayer-testnet -p 127.0.0.1:5432:5432 docker.io/library/postgres
+```
+
+Notice a few things:
+
+1. This command doesn't need root
+2. The database is running in trust mode, which means no username or password required
+3. The command line argument `--rm` is used, so that the container will be deleted on exit
+4. The database name is `mintlayer-testnet`, which is a standard used in the form of `mintlayer-`, followed by the network name (mainnet, testnet, etc)
```

Comment:
> which is a standard used in the form of `mintlayer-`

This part sounds off. Perhaps it's better to say something like:
"This is the default database name that api server will use for testnet. In general, the default database name is `mintlayer-`, followed by the network name (mainnet, testnet, etc)"

Reply:
A little longer, but OK. Done.